tests/test_synthetic_volume.py

```diff
@@ -14,127 +14,120 @@
     TSymmetricVolume,
 )
 
-
-class Base:
-    def setUp(self):
-        self.dtype = np.float32
-        self.C = 1
-        self.seed = 0
-        vol_kwargs = dict(
-            L=self.L,
-            C=self.C,
-            seed=self.seed,
-            dtype=self.dtype,
-        )
-        if hasattr(self, "order"):
-            vol_kwargs["order"] = self.order
-
-        self.vol_obj = self.vol_class(**vol_kwargs)
-        self.vol = self.vol_obj.generate()
-
-    def testVolumeRepr(self):
-        """Test Synthetic Volume repr"""
-        self.assertTrue(
-            repr(self.vol_obj).startswith(f"{self.vol_obj.__class__.__name__}")
+# dtype fixture to pass into volume fixture.
+DTYPES = [np.float32, pytest.param(np.float64, marks=pytest.mark.expensive)]
+
+
+@pytest.fixture(params=DTYPES)
+def dtype_fixture(request):
+    dtype = request.param
+    return dtype
+
+
+# Parameter combinations for testing SyntheticVolumes with cyclic and dihedral symmetry.
+# Each tuple represents (volume class, resolution in pixels, cyclic order)
+PARAMS_Cn_Dn = [
+    (CnSymmetricVolume, 20, 2),
+    (CnSymmetricVolume, 21, 2),
+    (CnSymmetricVolume, 30, 3),
+    (CnSymmetricVolume, 31, 3),
+    (CnSymmetricVolume, 40, 4),
+    (CnSymmetricVolume, 41, 4),
+    (CnSymmetricVolume, 52, 5),
+    (CnSymmetricVolume, 53, 5),
+    (CnSymmetricVolume, 64, 6),
+    (CnSymmetricVolume, 65, 6),
+    (DnSymmetricVolume, 20, 2),
+    (DnSymmetricVolume, 21, 2),
+    (DnSymmetricVolume, 40, 3),
+    (DnSymmetricVolume, 41, 3),
+    (DnSymmetricVolume, 42, 4),
+    (DnSymmetricVolume, 43, 4),
+    (DnSymmetricVolume, 55, 5),
+    (DnSymmetricVolume, 56, 5),
+    (DnSymmetricVolume, 64, 6),
+    (DnSymmetricVolume, 65, 6),
+]
```

Comment:
Note that you could code these out using `resolution%10`, but I wouldn't bother, would end up being similar amount of lines by the time you add two additional lists.

Reply:
noted!